pkg/utils/node/node.go

```diff
@@ -64,6 +66,23 @@ func GetProvisionablePods(ctx context.Context, kubeClient client.Client) ([]*v1.
 	}), nil
 }
 
+// GetVolumeAttachments grabs all attached volumeAttachments to passed node
+func GetVolumeAttachments(ctx context.Context, kubeClient client.Client, node *v1.Node) ([]*storagev1.VolumeAttachment, error) {
```

Comment:
I'm wondering if rather than getting all `VolumeAttachment` objects associated with the node, if we should only get `VolumeAttachment` objects associated with the drainable pods on the nodes. We know that any others won't be able to detach successfully until the node is terminating, and they could potentially be detached during graceful shutdown (if configured). Seems like it should be reasonable to look at all PVCs on drainable pods and map them to the correct `VolumeAttachment`, not sure about inline volumes though.

Reply:
Great call!

Regarding Inline: Not sure if we have to worry about inline volumes because they should only be used when lifecycle of volume matches that of pod:

```
A CSI driver is not suitable for inline use when:
- The volume needs to persist longer than the lifecycle of a pod
```

src: https://kubernetes.io/blog/2022/08/29/csi-inline-volumes-ga/

Reply:
Just to confirm, since the lifetime is tied together does that mean we know the VolumeAttachment will be deleted once the pod is terminated? Or is it still deleted asynchronously after?

Reply:
I am pretty sure pods with inline volumes never even have volume attachment objects that karpenter waitForVolumeDetaches could get stuck on. 

I'll double check this tomorrow, but I came to this belief from reading https://kubernetes.io/blog/2020/01/21/csi-ephemeral-inline-volumes/ and the https://kubernetes-csi.github.io/docs/ephemeral-local-volumes.html#implementing-csi-ephemeral-inline-support mentioning that "ephemeral volumes will not be created using the Controller Service, but the Node Service [and therefore kubelet], instead."